app/views/articles/_full_comment_area.html.erb

```diff
@@ -1,8 +1,9 @@
-<% cache("whole-comment-area-#{@article.id}-#{@article.last_comment_at}-#{@article.show_comments}", expires_in: 2.hours) do %>
+<% cache("whole-comment-area-#{@article.id}-#{@article.last_comment_at}-#{@article.show_comments}-#{@discussion_lock&.updated_at}", expires_in: 2.hours) do %>
```

Comment:
Typically, discussion locks won't be updated unless it's via a console or in the admin panel (not created yet). But this logic should still work in terms of looking up the cache.